jerry-ext/module/module.c

```diff
@@ -137,33 +131,52 @@ jerryx_module_add_to_cache (jerry_value_t cache, /**< cache to which to add the
 static const jerry_char_t *on_resolve_absent = (jerry_char_t *) "Module on_resolve () must not be NULL";
 
 /**
+ * Declare the linker section where module definitions are stored.
+ */
+JERRYX_SECTION_DECLARE (jerryx_modules, jerryx_native_module_t)
+
+/**
  * Declare and define the default module resolver - one which examines what modules are defined in the above linker
  * section and loads one that matches the requested name, caching the result for subsequent requests using the context
  * data mechanism.
  */
-bool
-jerryx_module_native_resolver (const jerry_char_t *name, /**< name of the module */
+static bool
+jerryx_resolve_native_module (const jerry_value_t name, /**< name of the module */
                                jerry_value_t *result) /**< [out] where to put the resulting module instance */
 {
   int index;
   const jerryx_native_module_t *module_p = NULL;
 
+  bool return_value = false;
+
+  jerry_size_t c_module_name_size = jerry_get_utf8_string_size (name);
+  jerry_char_t c_module_name[c_module_name_size];
```

Comment:
While this allocation is simple in source code level, I always has doubts about its efficiency. You need to move a lot of data on the stack.

Reply:
So, should I use `jmem_heap_alloc_block_null_on_error()` instead?

Reply:
I am just asking what is its cost. We could introduce convenience stuff for them (I saw you already created some macros for it).